### PR TITLE
Collect metrics from aspnetcore eventcounters

### PR DESCRIPTION
**Pull request by kevingosse**
Note: `Microsoft-AspNetCore-Server-Kestrel` provider is only available on .NET 5

### CODE REVIEW

File: src/Datadog.Trace/RuntimeMetrics/RuntimeEventListener.cs [hunk 1]
```diff
@@ -23,16 +27,38 @@ internal class RuntimeEventListener : EventListener, IRuntimeMetricsListener
...
+                ["failed-requests"] = MetricsNames.FailedRequests,
+                ["total-requests"] = MetricsNames.TotalRequests,
+                ["request-queue-length"] = MetricsNames.RequestQueueLength,
+                ["current-connections"] = MetricsNames.CurrentConnections,
+                ["connection-queue-length"] = MetricsNames.ConnectionQueueLength,
+                ["total-connections"] = MetricsNames.TotalConnections
+            };
+        }
+
+        public RuntimeEventListener(IDogStatsd statsd, int delay)
```
Comment (lucaspimentel) [inline src/Datadog.Trace/RuntimeMetrics/RuntimeEventListener.cs:56]: Can you specify the units for `delay`? Maybe in the parameter name (like `_delayInSeconds`) or using xml comments to document the ctor? Or even switch the type to `TimeSpan`?

File: src/Datadog.Trace/RuntimeMetrics/RuntimeMetricsWriter.cs [hunk 2]
```diff
@@ -31,7 +31,7 @@ public RuntimeMetricsWriter(IDogStatsd statsd, int delay)
         {
         }
 
-        internal RuntimeMetricsWriter(IDogStatsd statsd, int delay, Func<IDogStatsd, IRuntimeMetricsListener> initializeListener)
+        internal RuntimeMetricsWriter(IDogStatsd statsd, int delay, Func<IDogStatsd, int, IRuntimeMetricsListener> initializeListener)
```
Comment (lucaspimentel) [inline src/Datadog.Trace/RuntimeMetrics/RuntimeMetricsWriter.cs:34]: Similar to previous comment: can we specify the units for `delay`?

**Review by lucaspimentel — APPROVED** [review]
Nice addition to runtime metrics. Left some nits about unclear delay units (seconds? milliseconds?)

File: src/Datadog.Trace/RuntimeMetrics/MetricsNames.cs [hunk 3]
```diff
@@ -28,5 +28,14 @@ internal static class MetricsNames
         public const string CpuUserTime = "runtime.dotnet.cpu.user";
         public const string CpuSystemTime = "runtime.dotnet.cpu.system";
         public const string CpuPercentage = "runtime.dotnet.cpu.percent";
+
+        public const string CurrentRequests = "runtime.dotnet.aspnetcore.requests.current";
```
Comment (zacharycmontoya) [inline src/Datadog.Trace/RuntimeMetrics/MetricsNames.cs:32]: nit: Can we prefix these variables with `AspNetCore`?

Comment (kevingosse) [reply src/Datadog.Trace/RuntimeMetrics/MetricsNames.cs]: Done in https://github.com/DataDog/dd-trace-dotnet/pull/1074/commits/5ed0ba4effebc9ea07000a7024aab83f1e8e1f88

File: src/Datadog.Trace/RuntimeMetrics/RuntimeEventListener.cs (hunk 1 above)
Comment (kevingosse) [reply src/Datadog.Trace/RuntimeMetrics/RuntimeEventListener.cs]: Done in https://github.com/DataDog/dd-trace-dotnet/pull/1074/commits/5ed0ba4effebc9ea07000a7024aab83f1e8e1f88

**Review by zacharycmontoya — APPROVED** [review]
LGTM

**Pull request merged**